packages/labs/src/components/select/suggest.tsx

```diff
@@ -0,0 +1,271 @@
+    public static displayName = "Blueprint.Suggest";
+
+    public static ofType<T>() {
+        return Suggest as new () => Suggest<T>;
+    }
+
+    public state: ISuggestState<T> = {
+        isOpen: false,
+        isTyping: false,
+        query: "",
+    };
+
+    private input: HTMLInputElement;
+    private TypedQueryList = QueryList.ofType<T>();
+    private queryList: QueryList<T>;
+    private refHandlers = {
+        input: (ref: HTMLInputElement) => this.input = ref,
+        queryList: (ref: QueryList<T>) => this.queryList = ref,
+    };
+
+    public render() {
+        // omit props specific to this component, spread the rest.
+        const {
+            itemRenderer,
+            inputProps,
+            noResults,
+            popoverProps,
+            ...restProps,
+        } = this.props;
+
+        return <this.TypedQueryList
+            {...restProps}
+            activeItem={this.state.activeItem}
+            onActiveItemChange={this.handleActiveItemChange}
+            onItemSelect={this.handleItemSelect}
+            query={this.state.query}
+            ref={this.refHandlers.queryList}
+            renderer={this.renderQueryList}
+        />;
+    }
+
+    public componentDidUpdate(_prevProps: ISuggestProps<T>, prevState: ISuggestState<T>) {
+        if (this.state.isOpen && !prevState.isOpen && this.queryList != null) {
+            this.queryList.scrollActiveItemIntoView();
+        }
+    }
+
+    private renderQueryList = (listProps: IQueryListRendererProps<T>) => {
+        // not using defaultProps cuz they're hard to type with generics (can't use <T> on static members)
+        const { inputValueRenderer, inputProps = {}, popoverProps = {} } = this.props;
+        const { isTyping, selectedItem, query } = this.state;
+        const { ref, ...htmlInputProps } = inputProps;
+        const { handleKeyDown, handleKeyUp } = listProps;
+        const inputValue: string = isTyping
+            ? query : selectedItem
+                ? inputValueRenderer(selectedItem) : "";
+
+        return (
+            <Popover
+                autoFocus={false}
+                enforceFocus={false}
+                isOpen={this.state.isOpen}
+                position={Position.BOTTOM_LEFT}
+                {...popoverProps}
+                className={classNames(listProps.className, popoverProps.className)}
+                onInteraction={this.handlePopoverInteraction}
+                popoverClassName={classNames(Classes.SELECT_POPOVER, popoverProps.popoverClassName)}
+                popoverWillOpen={this.handlePopoverWillOpen}
+                popoverDidOpen={this.handlePopoverDidOpen}
+                popoverWillClose={this.handlePopoverWillClose}
+            >
+                <div
+                    onKeyDown={this.getTargetKeyDownHandler(handleKeyDown)}
+                    onKeyUp={this.state.isOpen ? handleKeyUp : undefined}
+                    onClick={this.handleInputClick}
```

Comment:
i imagine these events could go directly on `InputGroup`, no need for an _extra_ wrapper element